.github/workflows/build.yml

```
@@ -0,0 +1,244 @@
+        env:
+          CURRENT_VERSION: skip
+          PROJECT_VERSION: skip
+        with:
+          common-mvn-flags: -Declipse.p2.mirrors=false -Dmaven.test.skip=true -Dcyclonedx.skip=false
+          artifactory-reader-role: private-reader
+
+      - name: Custom versioning
+        run: .github/scripts/set_maven_build_version.sh
+      - name: Vault
+        id: secrets
+        uses: SonarSource/vault-action-wrapper@320bd31b03e5dacaac6be51bbbb15adf7caccc32 # 3.1.0
+        with:
+          secrets: |
+            development/artifactory/token/SonarSource-reddeer-public-deployer username | ARTIFACTORY_DEPLOY_USERNAME;
+            development/artifactory/token/SonarSource-reddeer-public-deployer access_token | ARTIFACTORY_DEPLOY_ACCESS_TOKEN;
+
+      - uses: SonarSource/ci-github-actions/build-maven@v1 #dogfood
+        env:
+          ARTIFACTORY_DEPLOY_USERNAME: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_DEPLOY_USERNAME }}
+          ARTIFACTORY_DEPLOY_ACCESS_TOKEN: ${{ fromJSON(steps.secrets.outputs.vault).ARTIFACTORY_ACCESS_TOKEN }}
+        with:
+          deploy-pull-request: true
+          maven-args: -Dcommercial -Pdeploy-sonarsource
+          artifactory-reader-role: private-reader
+          sonar-platform: none
+
+      - name: Upload P2 Repository
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: p2-repository
+          path: site/target/org.eclipse.reddeer.site-*.zip
+
+  qa:
+    runs-on: github-ubuntu-latest-s
+    name: QA - ${{ matrix.bundle }}
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - bundle: org.eclipse.reddeer.common.test
+            profile: common-tests
+          - bundle: org.eclipse.reddeer.core.test
+            profile: core-tests
+          - bundle: org.eclipse.reddeer.direct.test
+            profile: direct-tests
+          - bundle: org.eclipse.reddeer.e4.workbench.test
+            profile: e4.workbench-tests
+          - bundle: org.eclipse.reddeer.gef.test
+            profile: gef-tests
+          - bundle: org.eclipse.reddeer.graphiti.test
+            profile: graphiti-tests
+          - bundle: org.eclipse.reddeer.jface.test
+            profile: jface-tests
+          - bundle: org.eclipse.reddeer.junit.test
+            profile: junit-tests
+          - bundle: org.eclipse.reddeer.requirements.test
+            profile: requirements-tests
+          - bundle: org.eclipse.reddeer.swt.test
+            profile: swt-tests
+          - bundle: org.eclipse.reddeer.ui.test
+            profile: ui-tests
+          - bundle: org.eclipse.reddeer.uiforms.test
+            profile: uiforms-tests
+          - bundle: org.eclipse.reddeer.workbench.test
+            profile: workbench-tests
+    env:
+      DISPLAY: :10
+      MAVEN_OPTS: -Xmx3072m
+    steps:
+      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
```

```suggestion
      - uses: actions/checkout@v5
```

Please apply the same to the other occurrences